includes/ManageWikiHooks.php

```diff
@@ -154,39 +154,39 @@ public static function onCreateWikiCreation( $dbname, $private ) {
 		}
 
 		if ( ManageWiki::checkSetup( 'namespaces' ) ) {
+			$defaultCanonicalNamespaces = (array)ManageWiki::defaultCanonicalNamespaces();
+
 			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
-			$namespaces = $wgCanonicalNamespaceNames + [ 0 => '<Main>' ];
 
-			foreach ( $namespaces as $id => $name ) {
-				if ( $id < 0 ) {
-					// Don't include imaginary
-					continue;
-				}
+			foreach ( $defaultCanonicalNamespaces as $newnamespace ) {
+				$namespacesarray = ManageWiki::defaultNamespaces( $newnamespace );
 
-				$matchedNSKeys = array_keys( $wgNamespaceAliases, $id );
 				$nsAliases = [];
-
-				foreach ( $matchedNSKeys as $o => $n ) {
-					$nsAliases[] = $n;
+				if ( is_array( $namespacesarray['ns_aliases'] ) ) {
```

Comment:
This whole thing is unnecessary if you don’t decide the json

Reply:
Thanks, done.